python-checks/src/main/java/org/sonar/python/checks/hotspots/HashingDataCheck.java

```diff
@@ -0,0 +1,195 @@
+      PythonGrammar.CLASSDEF);
+  }
+
+  @Override
+  public void visitNode(AstNode node) {
+    if (node.is(PythonGrammar.ATTRIBUTE_REF, PythonGrammar.ATOM)) {
+      checkQuestionableHashingAlgorithm(node);
+
+    } else if (node.is(PythonGrammar.EXPRESSION_STMT)) {
+      checkOverwriteDjangoHashers(node);
+
+    } else if (node.is(PythonGrammar.CLASSDEF)) {
+      checkCreatingCustomHasher(node);
+
+    } else if (node.is(PythonGrammar.CALL_EXPR)) {
+      super.visitNode(node);
+    }
+  }
+
+  @Override
+  protected boolean isException(AstNode callExpression) {
+    if (getQualifiedName(callExpression).equals("django.contrib.auth.hashers.make_password")) {
+      AstNode argList = callExpression.getFirstChild(PythonGrammar.ARGLIST);
+      if (argList != null) {
+        return argList.getChildren(PythonGrammar.ARGUMENT).size() == 1;
+      }
+    }
+    return false;
+  }
+
+  private void checkOverwriteDjangoHashers(AstNode expressionStatement) {
+    AstNode lhsExpression = getLHSExpression(expressionStatement);
+    if (lhsExpression == null) {
+      return;
+    }
+
+    // checks for settings.PASSWORD_HASHERS = value
+    AstNode attributeRef = lhsExpression.getFirstDescendant(PythonGrammar.ATTRIBUTE_REF);
+    if (attributeRef != null) {
+      AstNode atom = attributeRef.getFirstChild(PythonGrammar.ATOM);
+      if (atom != null && getQualifiedName(atom).equals("django.conf.settings") &&
+        attributeRef.getLastChild(PythonGrammar.NAME).getTokenValue().equals("PASSWORD_HASHERS")) {
+        addIssue(expressionStatement, MESSAGE);
+      }
+    }
+
+    // checks for PASSWORD_HASHERS = [] in a global_settings.py
+    if (getContext().pythonFile().fileName().equals("global_settings.py") &&
+      lhsExpression.getTokenValue().equals("PASSWORD_HASHERS")) {
+      addIssue(expressionStatement, MESSAGE);
+    }
+  }
+
+  private void checkQuestionableHashingAlgorithm(AstNode node) {
+    String qualifiedName = getQualifiedName(node);
+    if (qualifiedName.equals("cryptography.hazmat.primitives.hashes") && isHashesFunctionCall(node.getParent())) {
+      addIssue(node.getParent(), MESSAGE);
+    } else if (questionableHashlibAlgorithm.contains(qualifiedName) || questionablePasslibAlgorithm.contains(qualifiedName)) {
+      addIssue(node, MESSAGE);
+    }
+  }
+
+  private static boolean isHashesFunctionCall(@Nullable AstNode node) {
+    if (node != null && node.is(PythonGrammar.ATTRIBUTE_REF)) {
+      String propertyName = node.getLastChild(PythonGrammar.NAME).getTokenValue();
+      AstNode parent = node.getParent();
+      return propertyName.equals("Hash") && parent != null && parent.is(PythonGrammar.CALL_EXPR);
+    }
+    return false;
+  }
+
+  private static AstNode getLHSExpression(AstNode expressionStatement) {
+    AstNode assign = expressionStatement.getFirstChild(PythonPunctuator.ASSIGN);
+    if (assign != null) {
+      return expressionStatement.getFirstChild(PythonGrammar.TESTLIST_STAR_EXPR);
```

Comment:
what if there is parenthesis?

Reply:
In case of parenthesis, TESTLIST_STAR_EXPR will still be the first child of EXPRESSION_STMT. 
Parenthesis will be consumed lower in the grammar (in `ATOM` production rule) and at line 123 I'm taking the `firstDescendant(PythonGrammar.ATTRIBUTE_REF)`.
I'm adding a test case to cover this.

However I just realized I won't raise an issue in this case:
```
foo.bar, mySettings.PASSWORD_HASHERS = value # Noncompliant
```
I will change my implementation to cover this case